ckanext/dia/views.py

```diff
@@ -121,3 +127,27 @@ def edit_uri(uri_id):
             vars['error_summary'] = { 'Error': _('An unknown error occurred') }
 
     return tk.render('uris/edit.html', extra_vars=vars)
+
+
+api = Blueprint(u'dia-api', __name__, url_prefix=u'/dia-api')
+
+@api.route('/datapusher-health', methods=['GET', 'POST'])
+def datapusher_health():
+    a_day_ago = datetime.datetime.now() - datetime.timedelta(days=1)
+
+    query = model.Session.query(model.TaskStatus)\
+        .filter(_and_(
+            model.TaskStatus.state == 'error',
+            model.TaskStatus.task_type == 'datapusher',
+            model.TaskStatus.last_updated > a_day_ago
+        ))
+
+    headers = {"Content-Type": "application/json;charset=utf-8"}
+
+    if query.count() > 0:
+        data = [model_dictize.task_status_dictize(ts, {'task_status': ts}) for ts in query.all()]
+        response_msg = json.dumps(data)
+        return make_response((response_msg, 500, headers))
```

Comment:
Example response:
```
[{"id": "12d7b0bd-2992-40ac-9037-51ac0d35b8ac", "entity_id": "e91d0448-86ef-4462-a862-77b46a3ca6da", "entity_type": "resource", "task_type": "datapusher", "key": "datapusher", "value": "{}", "state": "error", "error": "{\"message\": \"Could not connect to DataPusher.\", \"details\": \"HTTPConnectionPool(host='datapusher', port=8800): Max retries exceeded with url: /job (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f8bc3946890>: Failed to establish a new connection: [Errno -2] Name or service not known'))\"}", "last_updated": "2024-07-05T04:58:32.846214"}]
```

Reply:
Just thinking if 200 is a better option even when returning datapusher errors 🤔 Would it be confusing if the there is an actual 500 error thrown somewhere in here? I can also see how 500 can be useful as the caller won't have to check the payload before raising alarm or something. 🤷‍♂️

Reply:
Yeah, I guess a monitoring platform should be able to check for the existence of `error` in the response payload, and potentially then we'd be able to have different alerts based on 500, 200 with error, 200 ok